python/src/trezorlib/cli/btc.py

```diff
@@ -254,34 +276,45 @@ def _append_descriptor_checksum(desc: str) -> str:
 def _get_descriptor(
     client: "TrezorClient",
     coin: Optional[str],
-    account: str,
-    script_type: messages.InputScriptType,
+    account: int,
+    purpose: Optional[int],
+    script_type: Optional[messages.InputScriptType],
     show_display: bool,
 ) -> str:
-    coin = coin or DEFAULT_COIN
+    if purpose is None:
+        if script_type is None:
+            script_type = messages.InputScriptType.SPENDADDRESS
+        purpose = SCRIPT_TYPE_TO_BIP_PURPOSES[script_type][0]
+    elif script_type is None:
+        script_type = BIP_PURPOSE_TO_DEFAULT_SCRIPT_TYPE[purpose]
+    else:
+        if purpose not in SCRIPT_TYPE_TO_BIP_PURPOSES[script_type]:
+            raise ValueError("Invalid script type for account type")
+
     if script_type == messages.InputScriptType.SPENDADDRESS:
-        acc_type = 44
         fmt = "pkh({})"
     elif script_type == messages.InputScriptType.SPENDP2SHWITNESS:
-        acc_type = 49
         fmt = "sh(wpkh({}))"
     elif script_type == messages.InputScriptType.SPENDWITNESS:
-        acc_type = 84
         fmt = "wpkh({})"
     elif script_type == messages.InputScriptType.SPENDTAPROOT:
-        acc_type = 86
         fmt = "tr({})"
     else:
-        raise ValueError("Unsupported account type")
+        raise ValueError("Unsupported script type")
 
+    coin = coin or DEFAULT_COIN
     if coin is None or coin == "Bitcoin":
```

Comment:
Technically `coin is None` is never possible at this point thanks to the line above

Reply:
https://github.com/trezor/trezor-firmware/pull/2774/commits/de77d594585934f4bce89782a357590694d12324